webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/types/InferringGravsearchTypeInspector.scala

```diff
@@ -186,7 +186,7 @@ class InferringGravsearchTypeInspector(nextInspector: Option[GravsearchTypeInspe
     /**
      * Infers the `knora-api:objectType` of a property if the property's IRI is used as a predicate.
      */
-    private class KnoraObjectTypeFromPropertyIriRule(nextRule: Option[InferenceRule]) extends InferenceRule(nextRule = nextRule) {
+    private class InferTypeOfPropertyFromItsIri(nextRule: Option[InferenceRule]) extends InferenceRule(nextRule = nextRule) {
```

Comment:
The problem here is to make it clear that we mean "infer type information about a property" and not "infer type information about the object of a property". I think `ObjectType` and `KnoraObjectType` are ambiguous, because "object type" could mean "the type of the object of a property". I think `TypeOfProperty` makes it clear that we're getting information about the property and not about its object. Does that make sense?

Reply:
ok, I see what you are trying to do. Yes, it makes sense.